update roadmap for 2024 (#2310)
monero-project/monero-site: resources/roadmap/index.md

                    <h2>2023</h2>
                    <div class="tabPanel-content info-block">
                        <h2>2023</h2>
                            <ul>
                                <h3 class="months"></h3>
                                    <li class="completed">{% t roadmap.genimprov %}</li>
                                <h3 class="months">{% t roadmap.dec %}</h3>
                                    <li class="completed">{% t roadmap.lws-subaddress %}</li>
                            </ul>
                    </div>
                    <label for="tab-6" tabindex="0"></label>
                    <input id="tab-6" type="radio" name="tabs" aria-hidden="true" checked>
                    <h2>2024</h2>
                    <div class="tabPanel-content info-block">
                        <h2>2024</h2>
                            <ul>
                                <h3 class="months">{% t roadmap.jan %}</h3>
                                    <li class="completed">{% t roadmap.xmrbch-atomicswap %}</li>
                                <h3 class="months">{% t roadmap.may %}</h3>

>                                     <li class="completed">{% t roadmap.xmr-eth-atomicswaps %}</li>

Reviewer: May 2023

                                    <li class="completed">{% t roadmap.genimprov %}</li>
                            </ul>
                    </div>
                    <label for="tab-6" tabindex="0"></label>
                    <input id="tab-6" type="radio" name="tabs" aria-hidden="true" checked>
                    <h2>2024</h2>
                    <div class="tabPanel-content info-block">
                        <h2>2024</h2>
                            <ul>
                                <h3 class="months">{% t roadmap.jan %}</h3>
                                    <li class="completed">{% t roadmap.xmrbch-atomicswap %}</li>
                                <h3 class="months">{% t roadmap.may %}</h3>

>                                     <li class="completed">{% t roadmap.haveno-mainnet %}</li>

Reviewer 2: ```suggestion
                                    <li class="completed"><a href="https://haveno.exchange">{% t roadmap.haveno-mainnet %}</a></li>
```

                            </ul>
                    </div>
                    <label for="tab-7" tabindex="0"></label>
                    <input id="tab-7" type="radio" name="tabs" aria-hidden="true">
                    <h2>{% t roadmap.future %}</h2>
                    <div class="tabPanel-content info-block">
                        <h2>{% t roadmap.future %}</h2>
                            <ul>
                                <h3 class="months">{% t roadmap.comingsoon %}</h3>
                                    <li class="ongoing"><a href="https://github.com/Rucknium/OSPEAD">{% t roadmap.ospead %}</a></li>
                                    <li class="ongoing">{% t roadmap.xmr-eth-atomicswaps %}</li>
                                    <li class="ongoing">{% t roadmap.bp++ %}</li>
                                    <li class="ongoing">{% t roadmap.lws-subaddress %}</li>
                                    <li class="ongoing">{% t roadmap.cuprate %}</li>

>                                     <li class="upcoming">{% t roadmap.tryptych %}</li>

Reviewer: Remove

Reviewer 2: seems sane. using this as a reference: see 'proposal closed' https://ccs.getmonero.org/proposals/cypherstack-sarang-triptych-research.html

                        <h2>{% t roadmap.future %}</h2>
                            <ul>
                                <h3 class="months">{% t roadmap.comingsoon %}</h3>
                                    <li class="ongoing"><a href="https://github.com/Rucknium/OSPEAD">{% t roadmap.ospead %}</a></li>
                                    <li class="ongoing">{% t roadmap.xmr-eth-atomicswaps %}</li>
                                    <li class="ongoing">{% t roadmap.bp++ %}</li>
                                    <li class="ongoing">{% t roadmap.cuprate %}</li>

>                                     <li class="upcoming"><a href="https://github.com/seraphis-migration/strategy/wiki">{% t roadmap.seraphis-jamtis %}</a></li>

Reviewer 2: ```suggestion
                                    <li class="upcoming"><a href="https://github.com/seraphis-migration/strategy/wiki/Our-Wallet-Rewrite-Project-After-Priority-Switched-From-Seraphis-to-FCMPs">{% t roadmap.seraphis-jamtis %}</a></li>
```
rbrunner has added a new entry to the seraphis wiki above. it explains the "Seraphis & Jamtis" -> "Seraphis codebase & Jamtis" shift.

Author: I believe we shouldn't link the FCMP for seraphis roadmap, instead we should link for FCMP roadmap itself

>                                     <li class="upcoming"><a href="https://www.getmonero.org/2024/04/27/fcmps.html">{% t roadmap.fcmp-plus %}</a></li>

Author: this one

Reviewer 2: Can you show me what you want to change/remove/keep? its not clear to me

Author: move ``<a href="https://github.com/seraphis-migration/strategy/wiki">`` line 243 from line 242

Reviewer 2: make these changes as @nahuhh seems to agree and we can review further. What will `Seraphis codebase and jamtis` link to then?

>                                     <li class="upcoming"><a href="https://www.getmonero.org/2024/04/27/fcmps.html">{% t roadmap.fcmp-plus %}</a></li>

Reviewer: ```suggestion
                                    <li class="upcoming"><a href="https://github.com/seraphis-migration/strategy/wiki/Our-Wallet-Rewrite-Project-After-Priority-Switched-From-Seraphis-to-FCMPs">{% t roadmap.seraphis-jamtis %}</a></li>
```

@HardenedSteel like this, right?

Author: yes correct like this one @plowsof

Reviewer 2: this + removal of the hyperlink on seraphis codebase+jamtis LGTM
